app/controllers/vaccination_records_controller.rb

```diff
@@ -107,4 +108,8 @@ def set_breadcrumb_items
   def set_return_to
     @return_to = params[:return_to] || patient_path(@patient)
   end
+
+  def patient_id_for_logging
+    VaccinationRecord.find(params[:id]).patient_id
```

Comment:
I think we should avoid `patient_id_for_logging` needing to fetch data from the session or database, it means each call to an action in this controller will require at least two calls to the database, when we know for example in this case that the `before_action :set_vaccination_record` is going to run anyway which would allow us to get the `patient_id` from there. Although it's a simple query, unnecessary calls to the database has the potential to reduce performance of the service.

I think we need to find a way where we can execute the `set_vaccination_record` action before we wrap the action to assign the tags.